File: CSE 241 Object Oriented Programming/HW4/Hex.cpp
#include <iostream>
#include "Hex.h"
#include <fstream>
#include <string>
using namespace std;

Hex::Hex(int size, char type) {
    if (size >= 5)this->size = size;
    this->type = type;
    createBoard();
}
Hex::Hex(char type, int size) {
    if (size >= 5)this->size = size;
    this->type = type;
    createBoard();
}
Hex::Hex(int size) {
    if (size >= 5)this->size = size;
    createBoard();
}
Hex::Hex(char type) {
    this->type = type;
    createBoard();
}
//deletes pointers
void Hex::deletethings() {
    if (hexCells != nullptr) {
        for (int i = 0; i < 26; i++) {
            delete[] hexCells[i];
        }
        delete[] hexCells;
        hexCells = nullptr;
    }
    
    delete[] xPoints;
    delete[] yPoints;
    xPoints = nullptr;
    yPoints = nullptr;
}
Hex::Hex(const Hex& game) {
    deletethings();
    createBoard();
    for (int i = 0; i < 26; i++) {
        for (int j = 0; j < 26; j++) {
            hexCells[i][j].setVal(game.hexCells[i][j].getVal());
        }
    }
    size = game.getsize();
    type = game.gettype();
    totalmarked = game.gettotalmarked();
    lastx = game.getlastx();
    lasty = game.getlasty();
    winCond = game.getwinCond();
    maxPointUser = game.getUserScore();
    maxPointPc = game.getPcScore();
    xPoints = new int[26 * 26];
    yPoints = new int[26 * 26];
    for (int i = 0; i < game.gettotalmarked(); i++) {
        xPoints[i] = game.getXPoints()[i];
        yPoints[i] = game.getYPoints()[i];
    }
}

//de const
Hex::~Hex() {
    deletethings();
}
//assigment operator
Hex Hex::operator = (const Hex& game) {
    deletethings();
    createBoard();
    for (int i = 0; i < 26; i++) {
        for (int j = 0; j < 26; j++) {
            hexCells[i][j].setVal(game.hexCells[i][j].getVal());
        }
    }
    size = game.getsize();
    type = game.gettype();
    totalmarked = game.gettotalmarked();
    lastx = game.getlastx();
    lasty = game.getlasty();
    winCond = game.getwinCond();
    maxPointUser = game.getUserScore();
    maxPointPc = game.getPcScore();
    xPoints = new int[26 * 26];
    yPoints = new int[26 * 26];
    for (int i = 0; i < game.gettotalmarked(); i++) {
        xPoints[i] = game.getXPoints()[i];
        yPoints[i] = game.getYPoints()[i];
    }
    return *this;
}

//play for computer
Hex::Cell Hex::play() {
    int tempx = lastx;
    int tempy = lasty;

    int prex = lastx;

    while (true) {
        if (tempx != size - 1)tempx++;
        while (tempx >= 0 && hexCells[tempx][tempy].getVal() != moves::empty)tempx--;
        if (tempx == -1) {
            tempx = prex;
            tempy++;
            if (tempy == size)tempy = 0;
            if (tempy == lasty)prex++;
        }
        else break;
    }
    Cell temp;
    temp.setVal(moves::user1);
    setlastx(tempx);
    setlasty(tempy);
    //hexCells[tempx][tempy].setVal(moves::user1);
    //int map[26][26] = { 0 };
    //checkMaxPointForPc(tempx, tempy, tempx, tempy, map);
    insertCord(tempx, tempy);
    cout << "computer played at:" << (char)(tempy + 'A') << " " << tempx + 1 << endl;
    totalmarked++;
    return temp;
}
//play function for users

Hex::Cell Hex::play(int x, char y) {
    Cell temp;
    int a = y - 'A';
    //cout << a << " " << x;
    if (totalmarked % 2 == 0)
        //hexCells[x][a].setVal(moves::user2);
        temp.setVal(moves::user2);
    else
        //hexCells[x][a].setVal(moves::user1);
        temp.setVal(moves::user1);
    insertCord(x, a);
    lastx = x;
    lasty = y - 'A';
    totalmarked++;
    return temp;
}

//displays board
void Hex::displayBoard()const {
    cout << " ";
    for (int i = 0; i < getsize(); i++)cout << char('A' + i) << " ";
    cout << endl;
    for (int i = 0; i < getsize(); i++) {
        cout << i + 1;
        if (i < 9) {
            for (int j = 0; j < i + 1; j++) {
                cout << " ";
            }
        }
        else {
            for (int j = 0; j < i; j++) {
                cout << " ";
            }
        }
        for (int j = 0; j < getsize(); j++) {
            cout << hexCells[i][j].getVal() << " ";
        }
        cout << endl;
    }
    if (gettype() == '1') {
        int a = getGameScore();
        cout <<"User Score :" << getGameScore() << endl;
    }
}
//delete cord from arrays and board
void Hex::deleteCord() {
    if (gettotalmarked() == 0)return;
    int x = getXPointsLast();
    int y = getYPointsLast();
    hexCells[x][y].setVal(moves::empty);
    settotalmarked(gettotalmarked() - 1);
}
//create board
void Hex::createBoard() {
    hexCells = new Cell * [26];
    for (int i = 0; i < 26; i++) {
        hexCells[i] = new Cell[26];
    }
}
//add cords to arrays
void Hex::insertCord(int x, int y) {
    xPoints[gettotalmarked()] = x;
    yPoints[gettotalmarked()] = y;
}
//undo
void Hex::operator--() {
    deleteCord();
}
//undo
void Hex::operator --(int) {
    deleteCord();
}
//compare
bool Hex::operator == (const Hex& a) {
    return a.gettotalmarked() == gettotalmarked();
}
//decide load or save 
void Hex::callLoadSave(char ch) {
    string name;
    cout << "please write file name\n";
    cin >> name;

    if (ch == 's')saveFile(name);
    else loadFile(name);
}

//computer vs player
bool Hex::gameForComputer() {
    while (true) {
        cout << *this;
        cout << "[q]-quit , [u] - undo movement , [s] - Save game ,[l] - Load game\n";
        cout << "please write num and char in given format(A 1) \n";
        int x, y;
        char c;
        cin >> c;
        if (c == 'q') return true;
        if (c == 'u') {
            (*this)--;
            (*this)--;
            continue;
        }
        if (c == 's' || c == 'l') {
            callLoadSave(c);
            continue;
        }
        cin >> x;
        y = c - 'A';
        x--;
        //Checks is movement avalaible
        if (y > size - 1 || y < 0 || x>size - 1 || x < 0 || hexCells[x][y].getVal() != moves::empty) {
            cout << "\nwrong ! Try again\n";
            continue;
        }
        Cell tempo = play(x, c);
        getBoard()[x][y] = tempo;
        int map[26][26] = { 0 };
        int xx = x, yy = y;
        checkMaxPointForUser(x, y, xx, yy, map);
        /*hexCells[x][y].setVal(moves::user2);
        totalmarked++;
        nonempty++;*/
        

        if (checkWin(moves::user2)) {
            cout << "\n player1 won\n";
            winCond = 1;
            cout << *this;
            return true;
        }
        Cell temp = play();    
        getBoard()[getlastx()][getlasty()] = temp;
        int mapo[26][26] = { 0 };
        int a = getlastx();
        int b = getlasty();
        checkMaxPointForPc(getlastx(), getlasty(), a, b, mapo);

        if (checkWin(moves::user1)) {
            cout << "\n Computer won\n";
            winCond = 2;
            cout << *this;
            return true;
        }
        if (totalmarked == (size * size)) {
            cout << "\n DRAW\n";
            return true;
        }
    }
}


//this is game for player versus player
bool Hex::gameForUser() {
    while (true) {
        cout << *this;
        cout << "[q]-quit , [u] - undo movement , [s] - Save game ,[l] - Load game\n";
        if (totalmarked % 2 == 0)
            cout << "Turn for 1 \nplease write num and char in given format(A 1) \n";
        else
            cout << "Turn for 2 \nplease write num and char in given format(A 1) \n";

        int x, y;
        char c;
        cin >> c;
        if (c == 'q') return true;
        if (c == 'u') {
            (*this)--;
            continue;
        }
        if (c == 's' || c == 'l') {
            callLoadSave(c);
            continue;
        }
        cin >> x;
        y = c - 'A';
        x--;

        //Checks is movement avalaible
        if (y > size - 1 || y < 0 || x>size - 1 || x < 0 || hexCells[x][y].getVal() != moves::empty) {
            cout << "\nwrong ! Try again\n";
            continue;
        }

        Cell tempo = play(x, c);
        getBoard()[x][y] = tempo;
        int map[26][26] = { 0 };
        int xx = x, yy = y;
        checkMaxPointForPc(x, y, xx, yy, map);
         
        if (checkWin(moves::user2)) {
            cout << "\n player1 won\n";
            winCond = 1;
            cout << *this;
            return true;
        }

        if (checkWin(moves::user1)) {
            cout << "\n player2 won\n";
            winCond = 2;
            cout << *this;
            return true;
        }

        if (totalmarked == (size * size)) {
            cout << "\n DRAW\n";
            return true;
        }

    }
}

//checks is the game ended
void Hex::isWin(char player, int x, int y, bool winCondition[2], int map[50][50]) {
    if (x >= size || y >= size || x < 0 || y < 0 || map[x][y] == 1 || (winCondition[0] && winCondition[1]))return;
    if (player == moves::user2 && hexCells[x][y].getVal() != moves::user2)return;
    if (player == moves::user1 && hexCells[x][y].getVal() != moves::user1)return;
    hexCells[x][y].setVal(hexCells[x][y].getVal() - ' ');
    if (player == moves::user2) {
        if (x == 0)winCondition[0] = true;
        if (x == size - 1)winCondition[1] = true;
    }
    if (player == moves::user1) {
        if (y == 0)winCondition[0] = true;
        if (y == size - 1)winCondition[1] = true;
    }
    map[x][y] = 1;

    isWin(player, x - 1, y, winCondition, map);
    isWin(player, x - 1, y + 1, winCondition, map);
    isWin(player, x, y + 1, winCondition, map);
    isWin(player, x + 1, y, winCondition, map);
    isWin(player, x + 1, y - 1, winCondition, map);
    isWin(player, x, y - 1, winCondition, map);
    if (winCondition[0] && winCondition[1]) return;
    hexCells[x][y].setVal(hexCells[x][y].getVal() + ' ');

}
// checks is anybody won
bool Hex::checkWin(char player) {
    for (int i = 0; i < size; i++) {
        int map[50][50] = { 0 };
        bool winCondition[2] = { false };
        if (player == moves::user2 && hexCells[0][i].getVal() == moves::user2) {
            isWin(player, 0, i, winCondition, map);
        }
        else if (hexCells[i][0].getVal() == moves::user1) {
            isWin(player, i, 0, winCondition, map);
        }
        if (winCondition[0] && winCondition[1])
            return true;
    }


    return false;
}

//saves game to file
void Hex::saveFile(string name) const{
    ofstream myfile;
    myfile.open(name);  
    myfile << *this;
    myfile.close();

}
//loading file if its exist from file.txt
void Hex::loadFile(string name) {
    xPoints = new int[26 * 26];
    yPoints = new int[26 * 26];

    //cout << name;
    ifstream myfile(name);
    if (myfile.is_open()) {
        myfile >> *this;
        myfile.close();
        cout << "\nsuccesfully loaded\n";
    }

    else cerr << "there is no loadable file sorry :(";
    

}


//decide game type and start game
void Hex::playGame() {
    if (winCond == 1) {
        cout << "\n player1 won\n";
        cout << *this;
        return;
    }

    else if (winCond == 2) {
        cout << "\n player2 or Computer won\n";
        cout << *this;
        return;
    }

    if (type == '1') {
        gameForComputer();
    }
    else {
        gameForUser();
    }
}

//max point for user
void Hex::checkMaxPointForUser(int x, int y, int& maxX, int& minX, int map[26][26]) {
    if (x >= getsize() || y >= getsize() || x < 0 || y < 0 || map[x][y] == 1 )return;
    if (hexCells[x][y].getVal() != moves::user2)return;

    maxX = max(x, maxX);
    minX = min(x, minX);
    
    setUserScore(max(getUserScore(), (maxX - minX)));

    map[x][y] = 1;

    checkMaxPointForUser( x - 1, y, maxX, minX, map);
    checkMaxPointForUser( x - 1, y + 1, maxX, minX, map);
    checkMaxPointForUser( x, y + 1, maxX, minX, map);
    checkMaxPointForUser( x + 1, y, maxX, minX, map);
    checkMaxPointForUser( x + 1, y - 1, maxX, minX, map);
    checkMaxPointForUser( x, y - 1, maxX, minX, map);


}


//max point for pc
void Hex::checkMaxPointForPc(int x, int y, int& maxX, int& minX, int map[26][26]) {
    if (x >= getsize() || y >= getsize() || x < 0 || y < 0 || map[x][y] == 1)return;
    if (hexCells[x][y].getVal() != moves::user1)return;

    maxX = max(y, maxX);
    minX = min(y, minX);

    setPcScore(max(getPcScore(), (maxX - minX)));

    map[x][y] = 1;

    checkMaxPointForUser(x - 1, y, maxX, minX, map);
    checkMaxPointForUser(x - 1, y + 1, maxX, minX, map);
    checkMaxPointForUser(x, y + 1, maxX, minX, map);
    checkMaxPointForUser(x + 1, y, maxX, minX, map);
    checkMaxPointForUser(x + 1, y - 1, maxX, minX, map);
    checkMaxPointForUser(x, y - 1, maxX, minX, map);
}

// return games score
int Hex::getGameScore()const {
    if (getwinCond() == 1) return getsize();
    return getUserScore()-getPcScore();
}

File: CSE 241 Object Oriented Programming/HW7/Hex.java

//package hw7;

import java.awt.Color;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import javax.swing.JOptionPane;

/**
 *
 * @author omera
 */
public class Hex implements hexGame{
    Scanner scanner  = new Scanner(System.in);
    int gameSaveNum = 1;
    
    public void setLastx(int lastx) {
        this.lastx = lastx;
    }

    public void setLasty(int lasty) {
        this.lasty = lasty;
    }
    
    int size = 26;
    char type = '1';
    List<List<Cell>> hexCells = new ArrayList<>();
    int totalmarked = 0;
    private int lastx = 0;
    private int lasty = 0;
    int winCond = 0;
    
    Hex(int size , char type, int gsn) {
        if(size >=5)this.size = size;
        this.type = type;
        gameSaveNum = gsn;
        createBoard();
    }
    Hex(char type , int size, int gsn) {
        if(size >=5)this.size = size;
        this.type = type;
        gameSaveNum = gsn;
        createBoard();
    }
    Hex(int size, int gsn) {
        if (size >= 5)this.size = size;
        gameSaveNum = gsn;
        createBoard();
    }
    Hex(char type, int gsn) {
        this.type = type;
        gameSaveNum = gsn;
        createBoard();
    }
    public void createBoard(){
        hexCells.clear();
        for (int i = 0; i < 26; i++) {
            List<Cell> tempCell = new ArrayList<Cell>();
            for (int j = 0; j < 26; j++) {
                tempCell.add(new Cell('.'));
            }
            hexCells.add(tempCell);
        }
    }
    //void saveFile();
    //void loadFile();
    public int getSize(){
        return size;
    }
    public void displayBoard(){
        System.out.print(" ");
        for (int i = 0; i < size; i++)
            System.out.print((char)('A' + i)+" ");
        System.out.println("");
        for (int i = 0; i < size; i++) {
            System.out.print(i+1);
            if (i < 9) {
                for (int j = 0; j < i + 1; j++) {
                    System.out.print(" ");
                }
            }
            else {
                for (int j = 0; j < i; j++) {
                    System.out.print(" ");
                }
            }
            for (int j = 0; j < size; j++) {
                System.out.print(hexCells.get(i).get(j).getVal()+" ");
            }
            System.out.println("");
        }
    }
    public void play(){
        int tempx = lastx;
        int tempy = lasty;

        int prex = lastx;

        while (true) {
            if (tempx != size - 1)tempx++;
            while (tempx >= 0 && hexCells.get(tempx).get(tempy).getVal() != '.')tempx--;
            if (tempx == -1) {
                tempx = prex;
                tempy++;
                if (tempy == size)tempy = 0;
                if (tempy == lasty)prex++;
            }
            else break;
        }
        hexCells.get(tempx).get(tempy).setVal('o');
        lastx = tempx;
        lasty = tempy;
        System.out.println("computer played at:" + (char)(tempy + 'A') + " " + (tempx + 1));
        totalmarked++;
    }
    public void play(int x,char y){
        int a = y - 'A';
        if(totalmarked %2 == 0)
            hexCells.get(x).get(a).setVal('x');
        else 
            hexCells.get(x).get(a).setVal('o');
        lastx = x;
        lasty = y - 'A';
        totalmarked++;
    
    }
    public void isWin(char player, int x, int y, boolean[] winCondition, int[][] map){
        if (x >= size || y >= size || x < 0 || y < 0 || map[x][y] == 1 || (winCondition[0] && winCondition[1]))return;
        if (player == 'x' && hexCells.get(x).get(y).getVal() != 'x')return;
        if (player == 'o' && hexCells.get(x).get(y).getVal() != 'o')return;
        hexCells.get(x).get(y).setVal((char) (hexCells.get(x).get(y).getVal()-' '));
        if (player == 'x') {
            if (x == 0)winCondition[0] = true;
            if (x == size - 1)winCondition[1] = true;
        }
        if (player == 'o') {
            if (y == 0)winCondition[0] = true;
            if (y == size - 1)winCondition[1] = true;
        }
        map[x][y] = 1;

        isWin(player, x - 1, y, winCondition, map);
        isWin(player, x - 1, y + 1, winCondition, map);
        isWin(player, x, y + 1, winCondition, map);
        isWin(player, x + 1, y, winCondition, map);
        isWin(player, x + 1, y - 1, winCondition, map);
        isWin(player, x, y - 1, winCondition, map);
        if (winCondition[0] && winCondition[1]) return;
        hexCells.get(x).get(y).setVal((char) (hexCells.get(x).get(y).getVal()+' '));
    }
    public boolean checkWin(char player){
        for (int i = 0; i < size; i++) {
            int[][] map = new int[26][26];
            boolean[] winCondition ={false,false} ;
            if (player == 'x' && hexCells.get(0).get(i).getVal() == 'x') {
                isWin(player, 0, i, winCondition, map);
            }
            else if (hexCells.get(i).get(0).getVal() == 'o') {
                isWin(player, i, 0, winCondition, map);
            }
            if (winCondition[0] && winCondition[1])
                return true;
            
        }
    return false;
    }
    public void playGame(){
         if(winCond == 1){
            System.out.println("\n player1 won");
            displayBoard();
            return;
        }

        else if(winCond == 2){
            System.out.println("\n player1 or computer won");
            displayBoard();
            return;
        }

        displayBoard();

        if (type == '1') {
            gameForComputer();
        }
        else {
            gameForUser();
        }
    }
    public int totalMarked() { 
        return totalmarked; 
    }
    public boolean gameForUser(){
        while (true) {
            System.out.println("if you want quit type 'q'");
            if (totalmarked % 2 == 0)
                System.out.println("Turn for 1 please write num and char in given format(A 1)");
            else
                System.out.println("Turn for 2 please write num and char in given format(A 1)");

            int x, y;
            char c;
            c = scanner.next().charAt(0);
            if (c == 'q') return true;
            x = scanner.nextInt();
            y = c - 'A';
            x--;
            //Checks is movement avalaible
            if (y > size - 1 || y < 0 || x>size - 1 || x < 0 || hexCells.get(x).get(y).getVal() != '.') {
                System.out.println("\nwrong ! Try again");
                continue;
            }

            play(x, (char)(y + 'A'));

            displayBoard();
            if (totalmarked % 2 == 0) {
                if (checkWin('x')) {
                    System.out.println("\n player1 won");
                    winCond = 1;
                    displayBoard();
                    return true;
                }
            }

            else {
                if (checkWin('o')) {
                    System.out.println("\n player2 won");
                    winCond = 2;
                    displayBoard();
                    return true;
                }
            }
            if (totalmarked == (size * size)) {
                System.out.println("\n DRAW!!");
                return true;
            }

        }
    }

    public void setSize(int size) {
        this.size = size;
    }

    public void setType(char type) {
        this.type = type;
    }
    public boolean gameForComputer(){
        while (true) {
            System.out.println("if you want quit type 'q'");
            System.out.println("please write num and char in given format(A 1)");
            int x, y;
            char c;
            c = scanner.next().charAt(0);
            if (c == 'q') return true;
            x = scanner.nextInt();
            y = c - 'A';
            x--;
            //Checks is movement avalaible
            if (y > size - 1 || y < 0 || x>size - 1 || x < 0 || hexCells.get(x).get(y).getVal() != '.') {
                System.out.println("\nwrong ! Try again");
                continue;
            }
            play(x, c);
            displayBoard();

            if (checkWin('x')) {
                System.out.println("\n player1 won");
                winCond = 1;
                displayBoard();
                return true;
            }
            play();
            displayBoard();
            if (checkWin('o')) {
                System.out.println("\n computer won");
                winCond = 2;
                displayBoard();
                return true;
            }
            if (totalmarked == (size * size)) {
                System.out.println("\n DRAW!!");
                return true;
            }
        }
        
    }
    public boolean compare(Hex a) {
        return totalmarked > a.totalmarked;
    }  

    public int getLastx() {
        return lastx;
    }

    public int getLasty() {
        return lasty;
    }

    public void reset(){
        createBoard();
        HW7.mark=0;
        for (int i = 0 ; i < gameScreen.arr.length;i++){
            for (int j =0;j<gameScreen.arr.length;j++){
                gameScreen.arr[i][j].setBackground(Color.darkGray);
                hexCells.get(i).get(j).setVal('.');
            }
        }
        HW7.moves.clear();
    }
    
    public void save() throws IOException{
        String filename = JOptionPane.showInputDialog("File Name :");
        try {
            FileWriter myWriter = new FileWriter(filename);
            int size = gameScreen.arr.length;
            for(int i =0;i<26;i++){
                for(int j = 0;j<26;j++){
                    if(size <=i || size <=j)
                        myWriter.write(".");
                    else if(gameScreen.arr[i][j].getBackground() == Color.RED)
                        myWriter.write("x");
                    else if(gameScreen.arr[i][j].getBackground() == Color.blue)
                        myWriter.write("o");
                    else
                        myWriter.write(".");
                      
                    //myWriter.write(hexCells.get(i).get(j).getVal());
                }
                myWriter.write("\n");
            }
            myWriter.write(size+"\n");
            myWriter.write(type+"\n");
            myWriter.write(lastx+"\n");
            myWriter.write(lasty+"\n");
            myWriter.write(totalmarked+"\n");
            myWriter.write(winCond+"\n");
            for(int i = 0;i< HW7.moves.size();i++){
               myWriter.write(HW7.moves.get(i).getX()+" "+HW7.moves.get(i).getY()); 
               if(i != HW7.moves.size()-1)
                   myWriter.write("\n");
            }
            myWriter.close();
        } 
        catch (Exception e) {
            System.out.println("An error occurred.");
            e.printStackTrace();
        }
    }
    
    public void undo(){
        if(HW7.moves.size() == 0) {
            JOptionPane.showMessageDialog(null,"There is no move", "error",JOptionPane.ERROR_MESSAGE);
            return;
        }
        int x = HW7.moves.get(HW7.moves.size()-1).getX();
        int y = HW7.moves.get(HW7.moves.size()-1).getY();
        
        gameScreen.arr[x][y].setBackground(Color.darkGray);
        hexCells.get(x).get(y).setVal('.');
        HW7.mark--;
        HW7.moves.remove(HW7.moves.size()-1);
        
    }
    private int toInt(String s){
        int res=0;
        for(int i = 0;i< s.length();i++){
            if(s.charAt(i) == '\n' || s.charAt(i)=='\0') break;
            res*=10;
            res+=s.charAt(i)-'0';
        }
        return res;
    }
    
    public void load(){
        HW7.moves.clear();
        String filename = JOptionPane.showInputDialog("File Name :");
        try {
            
            File myObj = new File(filename);
            Scanner myReader = new Scanner(myObj);
            int counter = 0;
            while (myReader.hasNextLine()) {
                
                String data = myReader.nextLine();
                if(counter <26){
                    for(int i = 0;i<26;i++){
                        //System.out.println("sa");
                        hexCells.get(counter).get(i).setVal(data.charAt(i));
                    }
                }
                else if (counter == 26){
                    size = toInt(data);
                }
                else if(counter == 27){
                    type = data.charAt(0);
                }
                else if(counter == 28){
                    lastx = toInt(data);
                }
                else if(counter == 29){
                    lasty = toInt(data);
                }
                else if(counter == 30){
                    totalmarked = toInt(data);
                    HW7.mark = totalmarked;
                }
                else if(counter == 31){
                    winCond = toInt(data);
                }
                else{
                    
                    int t1 = 0;
                    int t2 = 0;
                    int i = 0;
                    while(data.charAt(i) != ' '){
                        //System.out.println(data+i);
                        t1*=10;
                        t1+=data.charAt(i)-'0';
                        i++;
                    }
                    i++;
                    while(i<data.length() && data.charAt(i) != '\n' && data.charAt(i) != '\0'){
                        //System.out.println(data+i);
                        t2*=10;
                        t2+=data.charAt(i)-'0';
                        i++;
                    }
                    HW7.moves.add(new point(t1, t2));
                    
                }
                counter++;
            }
            myReader.close();
     
            
            
        } 
        catch (FileNotFoundException e) {
            System.out.println("An error occurred.");
            e.printStackTrace();
        }
    }
    
    public char getType(){
        return type;
    }
    
}
